Gradle versionAsSpecified task is breaking when providing newVersion

When `VersionAsSpecified` task is called the exception is always thrown:

FAILURE: Build failed with an exception.

```
* What went wrong:
Configuration cache state could not be cached: input property `newVersion` of task `:versionAsSpecified` of type `org.gradle.api.DefaultTask`: error writing value of type 'org.gradle.api.internal.provider.OrElseProvider'
> No newVersion property provided! Please add the parameter -PnewVersion=<version> when running this task.

```

Need to fix this

--- a/gradle/plugins/src/main/kotlin/com.hedera.gradle.root.gradle.kts
+++ b/gradle/plugins/src/main/kotlin/com.hedera.gradle.root.gradle.kts
@@ -117,7 +117,7 @@ tasks.register("versionAsSpecified") {
 
     inputs.property("newVersion", providers.gradleProperty("newVersion").orElse(""))
 
-    if(inputs.properties["newVersion"] as String == "") {
+    if (inputs.properties["newVersion"] as String == "") {
         throw IllegalArgumentException(
             "No newVersion property provided! " +
                 "Please add the parameter -PnewVersion=<version> when running this task."

--- a/gradle/plugins/src/main/kotlin/com.hedera.gradle.root.gradle.kts
+++ b/gradle/plugins/src/main/kotlin/com.hedera.gradle.root.gradle.kts
@@ -115,19 +115,14 @@ tasks.register("versionAsSnapshot") {
 tasks.register("versionAsSpecified") {
     group = "versioning"
 
-    inputs.property(
-        "newVersion",
-        providers
-            .gradleProperty("newVersion")
-            .orElse(
-                provider {
-                    throw IllegalArgumentException(
-                        "No newVersion property provided! " +
-                            "Please add the parameter -PnewVersion=<version> when running this task."
-                    )
-                }
-            )
-    )
+    inputs.property("newVersion", providers.gradleProperty("newVersion").orElse("INVALID"))
+
+    if(inputs.properties["newVersion"] as String == "INVALID"){
+        throw IllegalArgumentException(
+            "No newVersion property provided! " +
+                "Please add the parameter -PnewVersion=<version> when running this task."
+        )
+    }
     outputs.file(layout.projectDirectory.versionTxt())
 
     doLast {

--- a/gradle/plugins/src/main/kotlin/com.hedera.gradle.root.gradle.kts
+++ b/gradle/plugins/src/main/kotlin/com.hedera.gradle.root.gradle.kts
@@ -42,18 +42,14 @@ tasks.register("githubVersionSummary") {
     group = "github"
 
     inputs.property("version", productVersion)
-    outputs.file(
-        providers
-            .environmentVariable("GITHUB_STEP_SUMMARY")
-            .orElse(
-                provider {
-                    throw IllegalArgumentException(
-                        "This task may only be run in a Github Actions CI environment! " +
-                            "Unable to locate the GITHUB_STEP_SUMMARY environment variable."
-                    )
-                }
-            )
-    )
+
+    if (!providers.environmentVariable("GITHUB_STEP_SUMMARY").isPresent) {
+        throw IllegalArgumentException(
+            "This task may only be run in a Github Actions CI environment! " +
+                "Unable to locate the GITHUB_STEP_SUMMARY environment variable."
+        )
+    }
+    outputs.file(providers.environmentVariable("GITHUB_STEP_SUMMARY"))
 
     doLast {
         generateProjectVersionReport(
@@ -115,9 +111,9 @@ tasks.register("versionAsSnapshot") {
 tasks.register("versionAsSpecified") {
     group = "versioning"
 
-    inputs.property("newVersion", providers.gradleProperty("newVersion").orElse(""))
+    inputs.property("newVersion", providers.gradleProperty("newVersion").orNull)
 
-    if (inputs.properties["newVersion"] as String == "") {
+    if (inputs.properties["newVersion"] == null) {
         throw IllegalArgumentException(
             "No newVersion property provided! " +
                 "Please add the parameter -PnewVersion=<version> when running this task."

--- a/gradle/plugins/src/main/kotlin/com.hedera.gradle.root.gradle.kts
+++ b/gradle/plugins/src/main/kotlin/com.hedera.gradle.root.gradle.kts
@@ -115,9 +115,9 @@ tasks.register("versionAsSnapshot") {
 tasks.register("versionAsSpecified") {
     group = "versioning"
 
-    inputs.property("newVersion", providers.gradleProperty("newVersion").orElse("INVALID"))
+    inputs.property("newVersion", providers.gradleProperty("newVersion").orElse(""))
 
-    if(inputs.properties["newVersion"] as String == "INVALID"){
+    if(inputs.properties["newVersion"] as String == "") {
         throw IllegalArgumentException(
             "No newVersion property provided! " +
                 "Please add the parameter -PnewVersion=<version> when running this task."

